LLVM compatibility patch of gcroots.ll for Julia 1.11 on LLVM 23, generated by test/upstream/derive.jl
from the rules (lines changed): masked-align (3), null-pointer-splat (3), splat (1)
--- a/gcroots.ll
+++ b/gcroots.ll
@@ -800,7 +800,7 @@
 ; TYPED: [[GEP0:%.*]] = getelementptr inbounds {} addrspace(10)*, {} addrspace(10)** %gcframe, i32 2
 ; TYPED: store {} addrspace(10)* %obj1, {} addrspace(10)** [[GEP0]]
 
-; OPAQUE: %arrayptrs = call <2 x ptr addrspace(13)> @llvm.masked.load.v2p13.p11(ptr addrspace(11) %arrayptrptr, i32 16, <2 x i1> <i1 true, i1 false>, <2 x ptr addrspace(13)> zeroinitializer)
+; OPAQUE: %arrayptrs = call <2 x ptr addrspace(13)> @llvm.masked.load.v2p13.p11(ptr addrspace(11) align 16 %arrayptrptr, <2 x i1> <i1 true, i1 false>, <2 x ptr addrspace(13)> {{zeroinitializer|splat \(ptr addrspace\(13\) null\)}})
 ; OPAQUE: [[GEP0:%.*]] = getelementptr inbounds ptr addrspace(10), ptr %gcframe, i32 2
 ; OPAQUE: store ptr addrspace(10) %obj1, ptr [[GEP0]]
 ;
@@ -828,7 +828,7 @@
 ; TYPED: [[GEP0:%.*]] = getelementptr inbounds {} addrspace(10)*, {} addrspace(10)** %gcframe, i32 2
 ; TYPED: store {} addrspace(10)* %obj1, {} addrspace(10)** [[GEP0]]
 
-; OPAQUE: %arrayptrs = call <2 x ptr addrspace(13)> @llvm.masked.gather.v2p13.v2p11(<2 x ptr addrspace(11)> %arrayptrptrs, i32 16, <2 x i1> <i1 true, i1 false>, <2 x ptr addrspace(13)> zeroinitializer)
+; OPAQUE: %arrayptrs = call <2 x ptr addrspace(13)> @llvm.masked.gather.v2p13.v2p11(<2 x ptr addrspace(11)> align 16 %arrayptrptrs, <2 x i1> <i1 true, i1 false>, <2 x ptr addrspace(13)> {{zeroinitializer|splat \(ptr addrspace\(13\) null\)}})
 ; OPAQUE: [[GEP0:%.*]] = getelementptr inbounds ptr addrspace(10), ptr %gcframe, i32 2
 ; OPAQUE: store ptr addrspace(10) %obj1, ptr [[GEP0]]
 ;
@@ -855,7 +855,7 @@
 ; TYPED: [[GEP0:%.*]] = getelementptr inbounds {} addrspace(10)*, {} addrspace(10)** %gcframe, i32 2
 ; TYPED: store {} addrspace(10)* %obj1, {} addrspace(10)** [[GEP0]]
 
-; OPAQUE: %arrayptrs = call <2 x ptr addrspace(13)> @llvm.masked.gather.v2p13.v2p11(<2 x ptr addrspace(11)> %arrayptrptrs, i32 16, <2 x i1> <i1 true, i1 true>, <2 x ptr addrspace(13)> zeroinitializer)
+; OPAQUE: %arrayptrs = call <2 x ptr addrspace(13)> @llvm.masked.gather.v2p13.v2p11(<2 x ptr addrspace(11)> align 16 %arrayptrptrs, <2 x i1> {{(<i1 true, i1 true>|splat \(i1 true\))}}, <2 x ptr addrspace(13)> {{zeroinitializer|splat \(ptr addrspace\(13\) null\)}})
 ; OPAQUE: [[GEP0:%.*]] = getelementptr inbounds ptr addrspace(10), ptr %gcframe, i32 2
 ; OPAQUE: store ptr addrspace(10) %obj1, ptr [[GEP0]]
 ;
